app/scripts/background.js

```diff
@@ -825,37 +827,113 @@ export function setupController(
     updateBadge,
   );
 
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_LIST_UPDATED,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_MARK_AS_READ,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.NOTIFICATIONS_STATE_CHANGE,
```

Comment:
Will this negate the above two listeners, assuming they result in notification state change?

Reply:
Wish it was a bit clearer, but notifications are not unified.
Following this event (through multiple files lol) points to the snap notifications, the other events points to the wallet notifications.

We are in discussion with the snaps team to better orchestrate and handle multiple notifications from different sources: Snap Notifications; Wallet Notifications; Feature Announcements; (future) Security Notifications.

Reply:
This is a good point though, I wonder if the snap notification controller has a nicer event to listen to instead of just state changed?

Reply:
Uhm, it seems that the Snap controller doesn't have an event related to receiving a notification, which is why I use the Notifications controller. 

@matthewwalsh0 , we actually have two controllers for managing notifications at the moment. One is the existing one, and the other is the one we introduced for receiving the new notifications. We built a new controller for the sake of implementation speed and because we have a whole new component of logic. In the future, we plan to work towards having a single controller (hopefully 🤞).

Reply:
Ah fair enough. Sure we can utilise the change event for now. We can discuss with snaps team if we can make specific events for notifications.